### PR TITLE
WIP: Add auto_complete_delay setting (defaulting to 500ms)

### PR DESCRIPTION
**Pull request by blueyed**
Implements: https://github.com/Shougo/deoplete.nvim/issues/204.

According to @justinmk it could use a timer in Python, but still needs to call Python then to trigger it (initially).  And it's not clear to me how to do this best.

### CODE REVIEW

**Comment by Shougo** [comment]
This is not I thought the implementation.

**Comment by tweekmonster** [comment]
@Shougo do you think you could reconsider even if it's a hard coded delay of 100ms or maybe a way to cancel previously triggered completions that are still pending?  I think a short delay would improve the responsiveness and avoid unnecessary completions.  I commented on deoplete-jedi about a problem that I thought this PR would help with: https://github.com/zchee/deoplete-jedi/issues/13#issuecomment-194684768

To summarize: Jedi is slow with broad completion searches and typing fast will cause deoplete-jedi to block subsequent completions until the previous ones are completed.

**Comment by justinmk** [comment]
@tweekmonster https://github.com/Shougo/deoplete.nvim/pull/212

**Comment by Shougo** [comment]
Yes, this feature is not rejected.

**Comment by tweekmonster** [comment]
Ah sorry. I misread. Thanks!
